src/neuroglancer/chunk_manager/frontend.ts

```diff
@@ -223,7 +223,7 @@ export class ChunkManager extends SharedObject {
       const newSource = new constructorFunction(this, options);
       newSource.initializeCounterpart(this.rpc!, {});
       return newSource;
-    });
+    }) as T;
```

Comment:
```
    ERROR in ./src/neuroglancer/chunk_manager/frontend.ts
    [tsl] ERROR in /Users/John/2017/fall/neuroglancer/src/neuroglancer/chunk_manager/frontend.ts(222,5)
          TS2322: Type 'RefCounted' is not assignable to type 'T'.
```
I suppose this will cause problems if one day a `constructorFunction` makes something that does not extend `ChunkSource`.